Makefile.tt

```diff
@@ -0,0 +1,57 @@
+include .env
+
+.PHONY: db/migrate db/rollback db/wait-for-postgres db/seed dev env/setup env/teardown test test/docker codebase codebase/fix
+
+db/migrate:
+	make db/wait-for-postgres
+	rails db:migrate
+
+db/rollback:
+	make db/wait-for-postgres
+	rails db:rollback
+
+db/wait-for-postgres:
+	$(shell DATABASE_URL=$(DATABASE_URL) ./bin/wait-for-postgres.sh)
+
+db/seed:
+	rails db:seed
+
+dev:
+	make install-dependencies
+	make env/setup
+	./bin/dev
+
+env/setup:
+	./bin/envsetup.sh
+	rails db:prepare
+
+env/teardown:  # this command will delete data
+	./bin/envteardown.sh
+
+generate-sprite:
+	./bin/svg-sprite
+
+install-dependencies:
+	bundle install
+	yarn install
+
+test:
+	docker-compose -f docker-compose.test.yml --project-name ewa-payroll-web-test up -d db redis
```

Comment:
I'm not sure we are bring this into the template, for a normal project we just need to run `rspec .....` no need to start a new `docker container for run the test and stop it`

This is not a small change so we would need to bring it as a discussion or RFC to explain why we need that, as only Flexi payroll had that till now

Reply:
@andyduong1920 usually, where is the DB used for tests? Is it in the same container? 🤔 
In flexi we used this way as it was conflicting with the application running on the local machine (i.e. you could not run a test when the application was running too).

Reply:
@andyduong1920 ,  not sure that I got your point 🤔 
Running tests with the docker container is already in the README:

https://github.com/nimblehq/rails-templates/blob/68125e59e5d4ef43e4d7502e800b44b529a5a87d/README.md.tt#L107

Reply:
@malparty 

1/Usually, where is the DB used for tests?
Same DB container @malparty in all Rails or Phoenix projects; we had no problem there.

I'm a bit surprised that when I can't run a test when the application is running, as it's connecting to different `databases` in each environment.

@tyrro 

It also mentioned to run without the `Docker way` 

![image](https://user-images.githubusercontent.com/11751745/230814204-614b7e9d-e105-48de-b9f7-0e59a3c52629.png)


And our template seems like a setting to run the `Test without docker way` on Local development, I think most developers used that, and we had no problem so far.

And you're missing understanding that point. Our problem here is, which DB container will we use for the test

1. Normally - same DB container as dev (different database name)
2. The mew Makefile introduce a new DB container for just the dev env

The new Makefile also still run the rspec from the local machine as well (non-Docker way) different as the Readme (where everything is inside the docker) FYI @tyrro.

![image](https://user-images.githubusercontent.com/11751745/230814599-a4b6ad57-0112-460a-b810-d407b50c777c.png)

---

I'm not against these changes, but I would like to know more about that, why we would need to run the test `in a different docker db container`.

This seems like introducing a new way to run the test to the template, we should have a discussion to inform the team as well.

And the last thing, I don't think the current `template` could support this new Makefile test could run, I believe we would need to modify the Database connection for the Test env.....

--

Summary:

1. Let's discuss why we need to have this, otherwise let's not introduce it into the template
2. If the team is ok with this, let's double check and modify the existing template to make sure the new Makefile test script could run 👌

Reply:
@andyduong1920 , Thanks for the detailed reply, removed `test`-related make commands in 7da2972fdb83938b24d3b3534e6c357167bd203e

@malparty , I tested the raw `rspec` command in a newly generated project as follows(all `addons` are selected)

<img width="1020" alt="Screen Shot 2023-04-11 at 4 28 20 PM" src="https://user-images.githubusercontent.com/39977805/231118297-7f5205db-764f-4507-9116-f8f1ebe087d4.png">

the contents of the `user_spec.rb` file were:

```ruby
require 'rails_helper'

RSpec.describe User, type: :model do
  it '1+1 returns 2' do
    p 'hey'
    expect(1 + 1).to eq 2
  end
end
```

I've to check with the Flexi project again to see if we are doing anything differently.

Reply:
I got your point now @andyduong1920, thanks for the explanation 🙌 

I just discovered/learned now that, in other projects, `tests` are run using the same container as the one that host the development DB. 
The `same` container seems OK to me. I think I configured it to another container as I got some conflicts but that was probably due to my lack of understanding of the different environments/ports/containers/DB/... 🙈 

So I think we should stick to the previous version (i.e. keep 1 single container to host both `_dev` and `_test` databases). 

One advantage of that it that we do NOT need to bring the container `down` at the end of tests (which I found super annoying in Flexi payroll, now that I use `rspec` command most of the time as it starts faster IF your container is already up).

Reply:
Oh, I see you removed the `make test` command — which resolved this thread too ;-)